source/MaterialXRender/ImageHandler.cpp

```diff
@@ -111,83 +216,65 @@ bool ImageHandler::saveImage(const FilePath& filePath,
     return false;
 }
 
-bool ImageHandler::acquireImage(const FilePath& filePath, ImageDesc& imageDesc, bool /*generateMipMaps*/, const Color4* /*fallbackColor*/)
+ImagePtr ImageHandler::acquireImage(const FilePath& filePath, bool, const Color4* fallbackColor, string* message)
 {
-    if (filePath.isEmpty())
-    {
-        return false;
-    }
-
-    string extension = filePath.getExtension();
+    FilePath foundFilePath =  _searchPath.find(filePath);
+    string extension = foundFilePath.getExtension();
     ImageLoaderMap::reverse_iterator iter;
     for (iter = _imageLoaders.rbegin(); iter != _imageLoaders.rend(); ++iter)
     {
         ImageLoaderPtr loader = iter->second;
         if (loader && loader->supportedExtensions().count(extension))
         {
-            bool acquired = loader->loadImage(filePath, imageDesc, getRestrictions());
-            if (acquired)
+            ImagePtr image = loader->loadImage(foundFilePath);
+            if (image)
             {
-                return true;
+                cacheImage(foundFilePath, image);
+                return image;
             }
         }
     }
-    return false;
-}
 
-bool ImageHandler::createColorImage(const Color4& color, ImageDesc& desc)
-{
-    unsigned int bufferSize = desc.width * desc.height * desc.channelCount;
-    if (bufferSize < 1)
+    if (message && !filePath.isEmpty())
     {
-        return false;
+        *message = string("Error loading image: ") + filePath.asString();
```

Comment:
Minor: I'm just curious what more we get from this as it just says something failed. Would we want to extend this to get failure message from a given image loader?

Reply:
For now, it’s just a way to recognize that the requested file failed to load, even though a valid image may have been returned using the constant-color fallback logic.  For the future, I like the idea of enhancing this error message with additional details from the loader.